src/commands/ssh/startSsh.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfigResource, User } from 'azure-arm-website/lib/models';
+import * as portfinder from 'portfinder';
+import * as vscode from 'vscode';
+import { SiteClient, TunnelProxy } from 'vscode-azureappservice';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { WebAppTreeItem } from '../../explorer/WebAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { delay } from '../../utils/delay';
+import * as remoteDebug from '../remoteDebug/remoteDebugCommon';
+
+export type sshTerminal = {
+    running: boolean,
+    terminal: vscode.Terminal | undefined
+};
+
+export const sshSessionsMap: Map<string, sshTerminal> = new Map();
+
+export async function startSsh(node?: SiteTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+
+    const currentSshTerminal: sshTerminal | undefined = sshSessionsMap.get(node.root.client.fullName);
+    if (currentSshTerminal && currentSshTerminal.running) {
+        throw new Error(`Azure SSH is currently starting or already started for "${node.root.client.fullName}".`);
```

Comment:
I'm not sure your map works in all cases. For example, if I send `ctrl+c` to the ssh terminal - wouldn't that exit me out of the ssh? But I believe your map would say I'm still connected and wouldn't let me connect again.

Reply:
Yeah, this is true.  The proxy is still open in this case so you could reconnect manually, but I'll try to figure out how to handle that scenario.

Reply:
If it gets complicated, feel free to file an issue and handle after this PR